cmd/cstor-pool-mgmt/controller/new-pool-controller/handler.go

```diff
@@ -0,0 +1,244 @@
+
+	uid := string(csp.GetUID())
+	importedPool := pool.ImportedCStorPools[uid]
+
+	// TODO
+	// In which scenario it is possible??
+	if importedPool != nil {
+		c.recorder.Event(importedPool,
+			corev1.EventTypeWarning,
+			string(common.AlreadyPresent),
+			string(common.MessageResourceAlreadyPresent))
+		return nil
+	}
+
+	// take a lock for common
+	// TODO move return to label and unlock
+	common.SyncResources.Mux.Lock()
+
+	status, isImported = pool.IsPoolImported(csp, true /* wait if Zrepl is initializing */)
+	if isImported {
+		c.recorder.Event(csp,
+			corev1.EventTypeNormal,
+			string(common.AlreadyPresent),
+			string(common.MessageResourceAlreadyPresent))
+		goto updatestatus
+	}
+
+	// try to import pool
+	status, isImported, err = pool.Import(csp)
+	if isImported {
+		if err != nil {
+			c.recorder.Event(csp,
+				corev1.EventTypeWarning,
+				string(common.FailureImported),
+				string(common.FailureImportOperations))
+			glog.Errorf("Failed to handle add event: import succeeded with failed operations %v", err.Error())
+		} else {
+			c.recorder.Event(csp,
+				corev1.EventTypeNormal,
+				string(common.SuccessImported),
+				string(common.MessageResourceImported))
+		}
+		goto updatestatus
+	}
+
+	//TODO: Why pool validation? should we consider zpool status during sync op only?
+
+	// IsInitStatus is to check if initial status of cstorpool object is `init`.
+	if pool.IsEmptyStatus(csp) || pool.IsPendingStatus(csp) {
+		err = pool.Create(csp)
+		if err != nil {
+			glog.Errorf("Pool creation failure: %v", string(csp.GetUID()))
+			c.recorder.Event(csp,
+				corev1.EventTypeWarning,
+				string(common.FailureCreate),
+				string(common.MessageResourceFailCreate))
+			status = string(apis.CStorPoolStatusOffline)
+			goto updatestatus
+		}
+		glog.Infof("Pool creation successful: %v", string(csp.GetUID()))
+		c.recorder.Event(csp,
+			corev1.EventTypeNormal,
+			string(common.SuccessCreated),
+			string(common.MessageResourceCreated))
+		status = string(apis.CStorPoolStatusOnline)
+		goto updatestatus
+	}
+	//TODO else set errored status
+
+updatestatus:
+	common.SyncResources.Mux.Unlock()
+	glog.Infof("TODO update status here  %s", status)
+	return err
+}
+
+func (c *CStorPoolController) destroyEventHandler(csp *apis2.CStorNPool) error {
+	// DeletePool is to delete cstor pool.
+	// It will also clear the label for relevant disk
+	err := pool.Delete(csp)
+	if err != nil {
+		c.recorder.Event(csp,
+			corev1.EventTypeWarning,
+			string(common.FailureDestroy),
+			string(common.MessageResourceFailDestroy))
+		// TODO
+		// update status here
+		// string(apis.CStorPoolStatusDeletionFailed), err
+		return err
+	}
+
+	// removeFinalizer is to remove finalizer of cStorPool resource.
+	err = c.removeFinalizer(csp)
+	if err != nil {
+		// TODO
+		// Object will exist. Let's set status as offline
+		return err
+	}
+	return nil
+}
+
+// syncEventHandler
+func (c *CStorPoolController) syncEventHandler(csp *apis2.CStorNPool) error {
```

Comment:
reconcilation required in syncEventHandler

Reply:
hmm